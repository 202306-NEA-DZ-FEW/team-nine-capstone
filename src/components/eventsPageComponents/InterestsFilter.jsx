import React from "react";
import { MdOutlineInterests } from "react-icons/md";

import { interestList } from "@/lib/interestsList";

function InterestsFilter({
    allCategories,
    handleInterest,
    setUpdatedInterests,
    updatedInterests,
    setAllCategoris,
}) {
    return (
        <div className='flex flex-col justify-center items-center gap-2'>
            <div className='flex flex-row justify-center mb-2 text-lg font-medium items-center w-48 border-b-2 gap-x-3 border-solid border-black '>
                {" "}
                Pick Your Interest <MdOutlineInterests />
            </div>
            <div
                onClick={() => {
                    setAllCategoris(!allCategories);
                }}
                className={`flex flex-row text-lg cursor-pointer hover:bg-orange-500 justify-center items-center gap-x-2 font-medium w-[90%] border-2 border-solid border-black rounded-md shadow-xl h-10 ${
                    allCategories
                        ? "bg-orange-500 transition-all scale-95 "
                        : ""
                }  `}
            >
                All
            </div>
            {interestList.map((interest) => (
                <div
                    key={interest.title}
                    className={`flex flex-row text-lg cursor-pointer transition-all hover:bg-orange-500 justify-center items-center gap-x-2 font-medium w-[90%] border-2 border-solid border-black rounded-md shadow-xl h-10 ${
                        !allCategories &&
                        updatedInterests.includes(interest.title)
                            ? "bg-orange-500 transition-all scale-95 "
                            : ""
                    } `}
                    onClick={() => {
                        setAllCategoris(null);
                        handleInterest(interest.title);
                    }}
                >
                    {interest.title}
                </div>
            ))}
        </div>
    );
}

export default InterestsFilter;
