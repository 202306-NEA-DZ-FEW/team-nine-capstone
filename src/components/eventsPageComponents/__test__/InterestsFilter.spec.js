import renderer from "react-test-renderer";

import InterestsFilter from "../InterestsFilter";

describe("InterestsFilter Snapshot", () => {
    it("should matches DOM Snapshot", () => {
        const tree = renderer.create(<InterestsFilter />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
