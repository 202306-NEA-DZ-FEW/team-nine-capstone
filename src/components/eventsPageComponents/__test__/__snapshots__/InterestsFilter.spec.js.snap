// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterestsFilter Snapshot should matches DOM Snapshot 1`] = `
<div
  className="flex flex-col justify-center items-center w-full gap-2 px-1 my-3"
>
  <div
    className=" gap-x-3 cursor-pointer bg-gray-200 rounded-xl w-[70%] h-10 shadow-lg  flex flex-row justify-center mb-2 text-lg font-medium items-center w-48  "
    onClick={[Function]}
  >
     
    eventList.Change interest
     
    <svg
      className="md:hidden"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M7.02 13c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zM13 13v8h8v-8h-8zm6 6h-4v-4h4v4zM7 2l-5 9h10L7 2zm0 4.12L8.6 9H5.4L7 6.12zM19.25 2.5c-1.06 0-1.81.56-2.25 1.17-.44-.61-1.19-1.17-2.25-1.17C13.19 2.5 12 3.78 12 5.25c0 2 2.42 3.42 5 5.75 2.58-2.33 5-3.75 5-5.75 0-1.47-1.19-2.75-2.75-2.75zM17 8.35c-1.45-1.22-3-2.4-3-3.1 0-.43.35-.75.75-.75.31 0 .52.17.73.37L17 6.3l1.52-1.43c.21-.2.42-.37.73-.37.4 0 .75.32.75.75 0 .7-1.55 1.88-3 3.1z"
      />
    </svg>
     
    <svg
      className="hidden md:block"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.178 19.569a.998.998 0 0 0 1.644 0l9-13A.999.999 0 0 0 21 5H3a1.002 1.002 0 0 0-.822 1.569l9 13z"
      />
    </svg>
  </div>
</div>
`;
